index.js

```diff
@@ -36,11 +37,40 @@ var reservedRe = /^\.+$/;
 var windowsReservedRe = /^(con|prn|aux|nul|com[0-9]|lpt[0-9])(\..*)?$/i;
 var windowsTrailingRe = /[\. ]+$/;
 
-function sanitize(input, replacement) {
+function containsInvalids(input, invalids) {
+  for (var i = 0; i < invalids.length; i++) {
+    if (input.indexOf(invalids[i]) !== -1) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function sanitize(input, replacement, invalids) {
   if (typeof input !== 'string') {
     throw new Error('Input must be string');
   }
-  var sanitized = input
+  if (typeof replacement === 'string' && invalids.indexOf(replacement) !== -1) {
+    throw new Error("The replacement string can't be part of options.additionalInvalidStrings or contain substrings which are part of options.additionalInvalidStrings!");
+  }
+  var sanitized = input;
+  var counter = 0;
+  while (containsInvalids(sanitized, invalids)) {
```

Comment:
There's quite a bit going on here, having a `for` loop running for every iteration of the `while`  loop, while running `indexOf` (which loops over the string). It'd be nice to avoid it. What if we just sanitize the return value of the replacement function instead?

Reply:
Sadly, this won't work. See my initial description for the PR, where I've specified why this is necessary. This won't be an issue for most inputs, because the while loop will only run once or twice and the for loop returns as soon as an invalid is found :)

Reply:
Fair enough :)